btrack/_localization.py

```diff
@@ -200,11 +241,39 @@ def segmentation_to_objects(
                 intensity_image=intens,
                 scale=scale,
                 use_weighted_centroid=USE_WEIGHTED,
+                assign_class_ID=assign_class_ID,
             )
 
             # concatenate the centroids
             centroids = _concat_centroids(centroids, _centroids)
 
+    elif DASK_INSTALLED:
+
+        if isinstance(segmentation, dask.array.core.Array):
+
+            if segmentation.ndim not in (3, 4):
+                raise ValueError("Segmentation array must have 3 or 4 dims.")
+
+            for frame in range(segmentation.shape[0]):
+                seg = segmentation[frame, ...].compute().astype(np.uint16)
+                intens = (
+                    intensity_image[frame, ...].compute().astype(np.uint16)
```

Comment:
Is there a reason why you set the dtype here?

Reply:
yeah it was loading them as 32 bit which was ott for segmentation imo...

Reply:
i would remove this `astype` from here - this is the kind of silent bug that causes problems for other people

Reply:
- [x] done